Source/WebKit/Shared/mac/WebCoreArgumentCodersMac.mm

```diff
@@ -271,4 +274,56 @@ static void encodeNSError(Encoder& encoder, NSError *nsError)
 }
 #endif
 
+template<typename Encoder>
+void ArgumentCoder<WebCore::AppKitControlSystemImage>::encode(Encoder& encoder, const WebCore::AppKitControlSystemImage& systemImage)
```

Comment:
Can we use Alex's new generated encoding/decoding thing here?

Reply:
The new mechanism doesn't support switching based on derived classes yet.

However, we can use the new approach to implement encoding/decoding of `AppKitControlSystemImage` subclasses. The one added in this patch doesn't have additional members, but the one to support `<progress>` will.